posthog/api/query.py

```diff
@@ -70,6 +71,11 @@ def get_throttles(self):
     @monitor(feature=Feature.QUERY, endpoint="query", method="POST")
     def create(self, request, *args, **kwargs) -> Response:
         data = self.get_model(request.data, QueryRequest)
+        if data.filters_override is not None:
+            data.query = apply_dashboard_filters_to_dict(
+                data.query.model_dump(), data.filters_override.model_dump(), self.team
+            )  # type: ignore
```

Comment:
`cast()` would better signal intentions than ignore

Reply:
might be missing a simple way to do this but 
the below doesn't work as data.query has to be one of 
<img width="932" alt="image" src="https://github.com/user-attachments/assets/ca339d87-8356-47b0-b23f-de82f7c9057b">
and the generated `schema.py` pulls all of the types as a Union [here](https://github.com/PostHog/posthog/blob/master/posthog/schema.py#L5872)
```
data.query = cast(
                QuerySchemaRoot,
                apply_dashboard_filters_to_dict(data.query.model_dump(), data.filters_override.model_dump(), self.team),
            )
```

Would be great if there was a way to cast it as below
```
data.query = cast(
                QueryRequest["query"],
                apply_dashboard_filters_to_dict(data.query.model_dump(), data.filters_override.model_dump(), self.team),
            )
```

Reply:
That's tough indeed… it's definitely a nit, so let's skip